src/Altinn.Platform/Altinn.Platform.Events/Events/Controllers/SubscriptionController.cs

```diff
@@ -0,0 +1,308 @@
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(eventsSubscription.Consumer))
+            {
+                message = "Missing event consumer";
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(eventsSubscription.AlternativeSubjectFilter)
+                && string.IsNullOrEmpty(eventsSubscription.SourceFilter))
+            {
+                message = "Source is required when subject is not defined";
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(eventsSubscription.CreatedBy))
+            {
+                message = "Invalid creator";
+                return false;
+            }
+
+            message = null;
+            return true;
+        }
+
+        /// <summary>
+        /// Validate that the identity (user, organization or org) is authorized to create subscriptions for given consumer. Currently
+        /// it needs to match. In future we need to add validation of business rules. (yet to be defined)
+        /// </summary>
+        private bool AuthorizeIdenityForConsumer(EventsSubscription eventsSubscription, out string message)
+        {
+            // First version require that 
+            if (!eventsSubscription.CreatedBy.Equals(eventsSubscription.Consumer))
+            {
+                message = "Not authorized to create a subscription on behalf of " + eventsSubscription.Consumer;
+                return false;
+            }
+
+            message = null;
+            return true;
+        }
+
+        /// <summary>
+        /// Validates that the identity (user, organization or org) is authorized to create subscriptions for a given subject.
+        /// Currently the subject needs to match the identity.  Org does not need subject.
+        /// </summary>
+        private bool AuthorizeSubjectForConsumer(EventsSubscription eventsSubscription, out string message)
+        {
+            if (!string.IsNullOrEmpty(eventsSubscription.AlternativeSubjectFilter) && !eventsSubscription.AlternativeSubjectFilter.Equals(eventsSubscription.Consumer))
+            {
+                message = "Not authorized to create a subscription with subject " + eventsSubscription.AlternativeSubjectFilter;
+                return false;
+            }
+
+            message = null;
+            return true;
+        }
+
+        private async Task<string> GetPartyFromAlternativeSubject(string alternativeSubject)
+        {
+            int partyId = 0;
+
+            if (alternativeSubject.StartsWith(OrganizationPrefix))
+            {
+                string orgNo = alternativeSubject.Replace(OrganizationPrefix, string.Empty);
+                partyId = await _registerService.PartyLookup(orgNo, null);
+            }
+            else if (alternativeSubject.StartsWith(PersonPrefix))
+            {
+                string persnoNo = alternativeSubject.Replace(OrganizationPrefix, string.Empty);
```

Comment:
Here I think its the PersonPrefix that should be replaced?

Reply:
Fixed and added some aditional logic